01_git/scissors/scissors.c

```diff
@@ -0,0 +1,73 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+static char *options[] = {
+    "rock\0",
+    "paper\0",
+    "scissors\0"
+};
+
+static char *options_user[] = {
+    ['r'] = "rock\0",
+    ['p'] = "paper\0",
+    ['s'] = "scissors\0"
+};
+
+enum {
+    ROCK = 0,
+    PAPER,
+    SCISSORS,
+};
+
+static char *game_logic[][3] = {
+    ['r'] = {
+        [ROCK] = "Draw",
+        [PAPER] = "I win: paper beats rock",
+        [SCISSORS] = "You win: rock beats scissors"
+    },
+    ['p'] = {
+        [ROCK] = "You win: paper beats rock",
+        [PAPER] = "Draw",
+        [SCISSORS] = "I win: scissors beats paper"
+    },
+    ['s'] = {
+        [ROCK] = "I win: rock beats scissors",
+        [PAPER] = "You win: scissors beats paper",
+        [SCISSORS] = "Draw"
+    },
+};
+
+int main()
+{
+    int ch, ch_user;
+    char buf[100];
+
+    // Use current time as seed
+    srand(time(0));   
+
+    printf("Please choose: rock (r) - paper (p) - scissors (s). q - exit.\n");
+
+    while(1) {
+
+        // wait for user choise
+        scanf("%s", buf);
```

Comment:
scanf("%s") is not safe.
it is possible to fix it but setting maximum number of chars.

Reply:
Fixed in https://github.com/Kernel-GL-HRK/gl_kernel_procamp_2021/pull/15/commits/b96e5a0237c2a6c2f7a8cc99f7905f573e7b7697